bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/SmartHomeDeviceHandler.java

```diff
@@ -185,62 +179,65 @@ public void updateChannelStates(List<SmartHomeBaseDevice> allDevices,
-                states = lastStates.get(applianceId);
-                if (states == null) {
-                    logger.trace("No old states array found for applianceId={}, trying to find old state.",
-                            applianceId);
-                    continue;
-                }
-                stateFound = true;
+            JsonArray states = applianceIdToCapabilityStates.getOrDefault(applianceId,
+                    lastStates.getOrDefault(applianceId, new JsonArray()));
+            if (states.size() == 0) {
+                logger.trace("No states array found for applianceId={}.", applianceId);
+                continue;
+            }
+            if (smartHomeBaseDevice.isGroup()) {
+                // for groups, store the last state of all devices
+                lastStates.put(applianceId, states);
             }
             logger.trace("Found states array={} for applianceId={}", states, applianceId);
 
-            if (firstDevice == null) {
-                firstDevice = shd;
-            }
             for (JsonElement stateElement : states) {
                 String stateJson = stateElement.getAsString();
                 if (stateJson.startsWith("{") && stateJson.endsWith("}")) {
                     JsonObject state = Objects.requireNonNull(gson.fromJson(stateJson, JsonObject.class));
-                    String interfaceName = Objects.requireNonNullElse(state.get("namespace"), JsonNull.INSTANCE)
-                            .getAsString();
-                    Objects.requireNonNull(mapInterfaceToStates.computeIfAbsent(interfaceName, k -> new ArrayList<>()))
-                            .add(state);
+                    JsonElement interfaceName = state.get("namespace");
+                    if (interfaceName != null) {
+                        Objects.requireNonNull(mapInterfaceToStates.computeIfAbsent(interfaceName.getAsString(),
+                                k -> new ArrayList<>())).add(state);
+                    }
                 }
             }
         }
 
+        if (mapInterfaceToStates.isEmpty()) {
+            logger.trace("Found no matching states.");
+            return;
+        }
+        logger.trace("mapInterfaceToState='{}'", mapInterfaceToStates);
+
         for (HandlerBase handlerBase : handlers.values()) {
             UpdateChannelResult result = new UpdateChannelResult();
             for (String interfaceName : handlerBase.getSupportedInterface()) {
                 List<JsonObject> stateList = mapInterfaceToStates.get(interfaceName);
                 if (stateList != null) {
                     try {
                         handlerBase.updateChannels(interfaceName, stateList, result);
-                    } catch (Exception e) {
+                    } catch (RuntimeException e) {
                         // We catch all exceptions, otherwise all other things are not updated!
```

Comment:
Not sure if it is a good idea to change to RuntimeException here. If anything else happens which is not expected, all other things will not be updated. This was the reason for catching Exception.

Reply:
But all exceptions that do not inherit from RuntimeException are checked exceptions and thus the compiler makes sure that they are catched.

Reply:
Ok, I see, this feature does not exist in C#, so I'am not familiar with it.